bigframes/functions/_function_session.py

```diff
@@ -400,6 +401,15 @@ def remote_function(
                 " For more details see https://cloud.google.com/functions/docs/securing/cmek#before_you_begin"
             )
 
+        if cloud_function_ingress_settings is None:
+            cloud_function_ingress_settings = "all"
+            msg = (
+                "The `cloud_function_ingress_settings` are set to 'all' by default, "
+                "which may change. Consider using 'internal-only' for enhanced security. "
+                "See https://cloud.google.com/functions/docs/networking/network-settings#ingress_settings for details."
+            )
+            warnings.warn(msg, category=UserWarning)
```

Comment:
Let's use [`FutureWarning`](https://docs.python.org/3/library/exceptions.html#FutureWarning) for this, since it's for a breaking change coming in the future. Also, the `stacklevel=2` is better than the default for helping users find the problem code using this (https://github.com/python/cpython/issues/88462#issuecomment-1093917557).

```suggestion
            warnings.warn(msg, category=FutureWarning, stacklevel=2)
```

Reply:
Also, TIL warn has added an option to be better than stacklevel which we can use in a few years when we drop Python 3.11 support:

> Changed in version 3.12: Added `skip_file_prefixes`.